remme/tp/pub_key.py

```
@@ -81,20 +85,24 @@ def _store_pub_key(self, context, signer_pubkey, transaction_payload):
             LOGGER.debug(f'entity_hash {transaction_payload.entity_hash}')
             raise InvalidTransaction('Entity hash or signature not a hex format')
 
-        # FIXME: For support PKCS1v15 and PSS
-        LOGGER.warn('HAZARD: Detecting padding for verification')
-        sigerr = 0
-        pkcs = padding.PKCS1v15()
-        pss = padding.PSS(mgf=padding.MGF1(hashes.SHA512()), salt_length=padding.PSS.MAX_LENGTH)
-        for _padding in (pkcs, pss):
+        if transaction_payload.public_key_type == NewPubKeyPayload.RSA:
+            if transaction_payload.rsa_signature_padding == NewPubKeyPayload.PSS:
+                _padding = padding.PSS(mgf=padding.MGF1(hashes.SHA512()), salt_length=padding.PSS.MAX_LENGTH)
+            elif transaction_payload.rsa_signature_padding == NewPubKeyPayload.PKCS1v15:
+                _padding = padding.PKCS1v15()
+
             try:
                 cert_signer_pubkey.verify(ehs_bytes, eh_bytes, _padding, hashes.SHA512())
                 LOGGER.warn('HAZARD: Padding found: %s', _padding.name)
             except InvalidSignature:
-                sigerr += 1
+                raise InvalidTransaction('Invalid signature')
+
+        elif transaction_payload.public_key_type == NewPubKeyPayload.ECDSA:
```

You should catch exceptions here and throw InvalidTransaction.